frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/PushAction.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.actions;
+
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.dvcs.repo.VcsRepositoryManager;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ * </ul>
+ */
+public class PushAction extends AnAction implements DumbAware {
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent e) {
+    super.update(e);
+
+    Project project = e.getProject();
+    Option<GitRepository> selectedVcsRepository = ActionUtils.getSelectedVcsRepository(e);
+
+    boolean isSomeRepositoryPresent = project != null && !VcsRepositoryManager.getInstance(project).getRepositories().isEmpty();
+    boolean isSelectedVcsRepositoryPresent = selectedVcsRepository.isDefined();
+
+    e.getPresentation().setEnabledAndVisible(isSomeRepositoryPresent && isSelectedVcsRepositoryPresent);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    Project project = ActionUtils.getProject(anActionEvent);
+
+    Option<GitRepository> selectedVcsRepository = ActionUtils.getSelectedVcsRepository(anActionEvent);
+    if (selectedVcsRepository.isDefined()) {
+      new VcsPushDialog(project, selectedVcsRepository.toJavaList(), /* currentRepo */ null).show();
```

Comment:
Ok... so this is going to do a push for all repositories, right?

Reply:
Are you going to change that to just the currently selected repo on this PR?

Reply:
Have a look at the dialog: 
![image](https://user-images.githubusercontent.com/19799111/80838147-a4b99f00-8bf8-11ea-9b6d-696a38b475a4.png)

As you can see there are checkboxes preceding names of repositories. A user can choose which of them he wants to make a push to. We are passing a single element list here containing the repository that is shown in the graph table at this moment. This repository is to be pre-selected.

Reply:
A branch to push is now passed here too! :rocket: